pandas_transform/pandas_transform.md

```diff
@@ -0,0 +1,990 @@
+* have an introductory level exposure to coding in [python](intro/to/python/module)
+* know how to run a [Jupyter notebook](link/here?).
+
+**Learning Objectives**
+
+@learning_objectives
+
+</div>
+
+## Lesson Preparation
+@sage
+
+You will be doing two types of hands-on coding as you work your way through this module. You do not need any particular preparation for the first way, interactive python cells. For the second way, Jupyter notebooks, you can choose to work in your browser where you will not be able to save your work, or in a few other ways where you will.
+
+**Interactive python cells**
+
+The first will be in cells powered by [SageMathCell](https://sagecell.sagemath.org/). For the most part, these will appear with some code already in them, and you can run that code by clicking the **Run python** button. You can also edit the code in these cells and run your own code.
+
+
+**Give it a try:**
+<div class="python_link">
+<lia-keep>
+<script type="text/x-sage">
+m = 3
+print(m+2)
+</script>
+</lia-keep>
+</div>
+
+The cells in this module are linked, meaning if you run code in two cells on the same page, the second cell will remember anything you defined in the first.
+
+<div class="python_link">
+<lia-keep>
+<script type="text/x-sage">
+n = m**3 + m  # a double asterisk indicates 3 is the exponent of m
+print(n)
+</script>
+</lia-keep>
+</div>
+
+You can change anything you want in either cell. Once you run that code using the **Run python** button, both cells will remember it.
+
+Code will not persist from one page to the next, and you can always refresh the page to return the code (and the stored memory of the cell) to its initial state.
+
+<div class = "important">
+These cells will compute everything you ask them to, but will only output what you explicitly request using the `print()` command.
+</div>
+
+**Jupyter notebooks** This section depends on what makes sense for the [Hands on activity](#hands-on-activity)
+
+The longer hands-on exercises in this module are in a Jupyter notebook. The notebook is embedded into the module and you are welcome to do the exercises there. However this will not save your work if you want to come back and see what you did in a few days. If you want that ability, you have two options:
+
+1. If you have python on your computer, you can download he notebook and run it yourself.
+
+2. Make and account somewhere to run it in the cloud?
+
+
+
+## The `pandas` Package
+
+The `pandas` [package](https://pandas.pydata.org/) lets you store, examine, and manipulate tabular data using python. Since many machine learning tools use python, it can be particularly useful to process tabular data in that same environment.
+
+<div class = "learnmore">
+Most people associate "pandas" with the large mammals [Ailuropoda melanoleuca](https://en.wikipedia.org/wiki/Giant_panda). The `pandas` package is actually a shortening of "panel data."
+</div>
+
+### Importing `pandas`
+
+**`import pandas`**
```

Comment:
I think having these "headers" that are pure code is a little confusing -- I thought at first that they should be run. Putting in suggestions for revisions, feel free to take/leave/edit as you see fit, of course.